chainer/training/extensions/parameter_statistics.py

```diff
@@ -0,0 +1,173 @@
+    iterable of :class:`~chainer.Link`s. If a link contains child links, the
+    statistics are collected separately for each child.
+
+    Statistics that can be collected and reporter using the current scope are
+    minimum and maximum values, means, standard deviations, percentiles and
+    zero counts.
+
+    Args:
+        links (~chainer.Link or iterable of ~chainer.Link): Links containing
+            the parameters to observe. The link is expected to have a ``name``
+            attribute which is used as a part of a key in the report.
+        trigger: Trigger that decides when to aggregate the results and report
+            the values.
+        report_params (bool): If ``True``, include parameter statistics in the
+            report such as weight and bias values.
+        report_param_grads (bool): If ``True``, include parameter statistics
+            in the report such as weight and bias gradients.
+        count_zeros (bool): If ``True``, count the number of zero elements and
+            include those statistics in the report. Else, do not compute the
+            number of zero elements.
+        prefix (str): Prefix to prepend to the report keys.
+        targets (iterable of tuples): Target parameters and their attributes
+            to consider in this extension.
+    """
+    default_name = 'parameter_statistics'
+    priority = extension.PRIORITY_WRITER
+
+    def __init__(self, links, trigger=(1, 'epoch'), report_params=True,
+                 report_param_grads=True, count_zeros=False, prefix=None):
+
+        if not isinstance(links, (list, tuple)):
+            links = links,
+        self._links = links
+
+        self._trigger = training.trigger.get_trigger(trigger)
+        self._count_zeros = count_zeros
+        self._prefix = prefix
+
+        self._attrs = []
+        if report_params:
+            self._attrs.append('data')
+        if report_param_grads:
+            self._attrs.append('grad')
+
+        self._statistic_functions = ('min', 'max', 'mean', 'std')
+        self._percentile_sigmas = (0.13, 2.28, 15.87, 50, 84.13, 97.72, 99.87)
+
+        self._summary = reporter.DictSummary()
+
+    def __call__(self, trainer):
+        """Execute the statistics extension.
+
+        Collect statistics for the current state of parameters.
+
+        Note that this method will merely update its statistic summary, unless
+        the internal trigger is fired. If the trigger is fired, the summary
+        will also be reported and then reset for the next accumulation.
+
+        Args:
+            trainer (~chainer.training.Trainer): Associated trainer that
+                invoked this extension.
+        """
+
+        for link in self._links:
+            link_name = getattr(link, 'name', 'None')
+            for param_name, param in link.namedparams():
+                for attr_name in self._attrs:
+                    prefix = '{}{}/{}/'.format(link_name,
```

Comment:
I don't think it is needed since the names obtained by [`namedparams()`](https://github.com/pfnet/chainer/blob/735f8b49342bf58f2ff730492fca4dba2c879aa5/chainer/link.py#L362-L363) always seem to start with a `/` already.